src/codeflare_sdk/cluster/cluster.py

```diff
@@ -415,25 +416,48 @@ def cluster_dashboard_uri(self) -> str:
         """
         Returns a string containing the cluster's dashboard URI.
         """
-        try:
-            config_check()
-            api_instance = client.NetworkingV1Api(api_config_handler())
-            ingresses = api_instance.list_namespaced_ingress(self.config.namespace)
-        except Exception as e:  # pragma no cover
-            return _kube_api_error_handling(e)
+        config_check()
+        if is_openshift_cluster():
+            try:
+                api_instance = client.CustomObjectsApi(api_config_handler())
+                routes = api_instance.list_namespaced_custom_object(
```

Comment:
Could this be turned into a get operation rather than a list? Maybe with a label selector if needed?